crates/core/src/table/builder.rs

```diff
@@ -261,7 +261,22 @@ impl DeltaTableBuilder {
     /// - [S3 options](https://docs.rs/object_store/latest/object_store/aws/enum.AmazonS3ConfigKey.html#variants)
     /// - [Google options](https://docs.rs/object_store/latest/object_store/gcp/enum.GoogleConfigKey.html#variants)
     pub fn with_storage_options(mut self, storage_options: HashMap<String, String>) -> Self {
-        self.storage_options = Some(storage_options);
+        self.storage_options = Some(
+            storage_options
+                .clone()
+                .into_iter()
+                .map(|(k, v)| {
+                    let needs_trim = v.starts_with("http://")
+                        || v.starts_with("https://")
+                        || k.to_lowercase().ends_with("_url");
```

Comment:
what are we checking here? i.e. what does a `_url` suffix signify? 

I'm probably just missing something obvious 😆.

Reply:
ah - from the tests i gather this is a convention we are introducing?

Should we document this behavior in a docstring somewhere?